src/shared/decodeImageFrame.js

```diff
@@ -138,15 +141,9 @@ function decodeImageFrame(
     throw new Error('decodePromise not defined');
   }
 
-  decodePromise
-    .then((imageFrame) => {
-      callbackFn(
-        postProcessDecodedPixels(imageFrame, options, start, decodeConfig)
-      );
-    })
-    .catch((err) => {
-      throw err;
-    });
+  const decodedFrame = await decodePromise;
```

Comment:
can you add error handling to this? it had it before

Reply:
It uses the implicit await error handling, which is actually functional, as opposed to the old error handling which ended up being a no-op.